webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/SettingsApi.scala

```diff
@@ -698,6 +707,86 @@ class SettingsApi(
     def get = configService.rudder_generation_continue_on_error()
     def set = (value : Boolean, _, _) => configService.set_rudder_generation_continue_on_error(value)
   }
+
+  object GetAllAuthorizedNetworks extends LiftApiModule0 {
+    override val schema = API.GetAllAuthorizedNetworks
+
+    val restExtractor = restExtractorService
+    //implicit val prettify = restExtractor.extractPrettify(req.params)
+    def process0(version: ApiVersion, path: ApiPath, req: Req, params: DefaultParams, authzToken: AuthzToken): LiftResponse = {
+      implicit val action = "getAllAuthorizedNetworks"
+      import com.normation.utils.Control.sequence
+      val result = for {
+        servers <- nodeInfoService.getAllSystemNodeIds()
+        authorizedNetworks <- sequence(servers)(nodeId => policyServerManagementService.getAuthorizedNetworks(nodeId).map((nodeId, _)))
+      } yield {
+        import net.liftweb.json.JsonDSL._
+        JArray(authorizedNetworks.toList.map {
+          case (nodeid, networks) =>
+            ("id" -> nodeid.value) ~ ("authorizedNetworks" -> networks.toList)
+        })
+      }
+      RestUtils.response(restExtractorService, "settings", None)(result, req, s"Could not get authorized networks")
+    }
+  }
+
+
+  object GetAuthorizedNetworks extends LiftApiModule {
+    override val schema = API.GetAuthorizedNetworks
+
+    val restExtractor = restExtractorService
+    //implicit val prettify = restExtractor.extractPrettify(req.params)
+    def process(version: ApiVersion, path: ApiPath, id: String, req: Req, params: DefaultParams, authzToken: AuthzToken): LiftResponse = {
+
+      implicit val action = "getAuthorizedNetworks"
+      val result = for {
+        nodeInfo <- nodeInfoService.getNodeInfo(NodeId(id))
+        isServer <- nodeInfo match {
+          case Some(info) if info.isPolicyServer =>
+            Full("I'm a policy server")
```

Comment:
Can you retrun `Full(())`, it will be simpler to translate to `zio` at some point in the future.

Reply:
Is there a problem with returning Unit here ?